jena-fuseki2/jena-fuseki-main/src/main/java/org/apache/jena/fuseki/main/sys/FusekiModules.java

```diff
@@ -18,67 +18,56 @@
 
 package org.apache.jena.fuseki.main.sys;
 
-import java.util.ArrayList;
 import java.util.List;
 import java.util.function.Consumer;
 
-import org.apache.jena.base.module.Subsystem;
-
-/** Registry of modules */
+/**
+ * List of {@linkplain FusekiModule Fuseki modules}.
+ * This is the immutable unit of a modules for a server.
+ * <p>
+ * @see FusekiModulesLoaded
+ */
 public class FusekiModules {
 
-    private static Object lock = new Object();
+    /** A Fuseki module with no members. */
+    public static FusekiModules empty = FusekiModules.create();
 
-    // Record of what is loaded.
-    private static List<FusekiModule> registry = null;
-
-    private static Subsystem<FusekiModule> subsystem = null;
+    /** Create a collection of Fuseki modules */
+    public static FusekiModules create(FusekiModule ... modules) {
+        return new FusekiModules(modules);
+    }
 
-    public static void load() {
-        if ( registry == null )
-            reload();
+    /** Create a collection of Fuseki modules */
+    public static FusekiModules create(List<FusekiModule> modules) {
+        return new FusekiModules(modules);
     }
 
-    public static void reload() {
-        registry = new ArrayList<>();
-        subsystem = new Subsystem<FusekiModule>(FusekiModule.class);
-        subsystem.initialize();
-        synchronized(lock) {
-            subsystem.forEach(registry::add);
-        }
+    private final List<FusekiModule> modules;
+
+    private FusekiModules(FusekiModule ... modules) {
+        this.modules = List.of(modules);
```

Comment:
NPE here if `modules` is `null` (coming from the public `#create` method).

Reply:
Done. Always good to check early - this is unlikely to be performance critical code.